data/state/accountsDB.go

```diff
@@ -856,9 +865,19 @@ func (adb *AccountsDB) SetStateCheckpoint(rootHash []byte) {
 		adb.mainTrie.SetCheckpoint(rootHash)
 		adb.snapshotUserAccountDataTrie(rootHash)
 		adb.mainTrie.ExitSnapshotMode()
+
+		adb.increaseNumCheckpoints()
 	}()
 }
 
+func (adb *AccountsDB) increaseNumCheckpoints() {
+	time.Sleep(batchWriteDelay)
```

Comment:
we depend upon a variable defined in config that can be easily changed without affecting this time.Sleep

Reply:
Refactored.